crates/bitwarden-wasm-internal/src/ssh.rs

```diff
@@ -0,0 +1,58 @@
+use bitwarden_ssh;
+use wasm_bindgen::prelude::*;
+
+#[wasm_bindgen]
+pub enum KeyAlgorithm {
+    Ed25519,
+    Rsa3072,
+    Rsa4096,
+}
+
+// impl conversion
+impl From<KeyAlgorithm> for bitwarden_ssh::KeyAlgorithm {
+    fn from(key_algorithm: KeyAlgorithm) -> Self {
+        match key_algorithm {
+            KeyAlgorithm::Ed25519 => bitwarden_ssh::KeyAlgorithm::Ed25519,
+            KeyAlgorithm::Rsa3072 => bitwarden_ssh::KeyAlgorithm::Rsa3072,
+            KeyAlgorithm::Rsa4096 => bitwarden_ssh::KeyAlgorithm::Rsa4096,
+        }
+    }
+}
+
+#[wasm_bindgen]
+pub struct SshKey {
+    private_key: String,
+    public_key: String,
+    key_fingerprint: String,
+}
+
+impl From<bitwarden_ssh::models::SshKey> for SshKey {
+    fn from(key: bitwarden_ssh::models::SshKey) -> Self {
+        SshKey {
+            private_key: key.private_key,
+            public_key: key.public_key,
+            key_fingerprint: key.key_fingerprint,
+        }
+    }
+}
+
+impl SshKey {
+    pub fn private_key(&self) -> &str {
+        &self.private_key
+    }
+
+    pub fn public_key(&self) -> &str {
+        &self.public_key
+    }
+
+    pub fn key_fingerprint(&self) -> &str {
+        &self.key_fingerprint
+    }
+}
```

Comment:
@coroiu Do we need to re-define the types? We should look into a way to conditionally apply `wasm_bindgen` in the original crates, similar to our uniffi flow.

Reply:
You should be able to use `#[cfg_attr(feature = "wasm", wasm_bindgen]` but ideally `bitwarden_ssh::models::SshKey` would implement `tsify` instead. See https://contributing.bitwarden.com/architecture/sdk/password-manager/web/interoperability#how-to-use-tsify